cachix/src/Cachix/Client/Push.hs

```diff
@@ -83,31 +84,38 @@ data PushParams m r = PushParams
   }
 
 data PushStrategy m r = PushStrategy
-  { -- | Called when a path is already in the cache.
+  { -- | An action to call when a path is already in the cache.
     onAlreadyPresent :: m r,
+    -- | An action to run on each push attempt.
     onAttempt :: RetryStatus -> Int64 -> m (),
+    -- | A conduit to inspect the NAR stream before compression.
+    -- This is useful for tracking progress and computing hashes. The conduit must not modify the stream.
+    onStream :: RetryStatus -> Int64 -> ConduitT ByteString ByteString (ResourceT m) (),
```

Comment:
```suggestion
    onUncompressedNARStream :: RetryStatus -> Int64 -> ConduitT ByteString ByteString (ResourceT m) (),
```

Preemptive disambiguation for when you want to show the network and/or compression speed (whichever is the bottleneck), using the compressed stream.

Reply:
I was thinking `onStream`/`onCompressedStream`, but this works too.